[Temp] Starter prompts

stephmilovic opened the pull request:
Temp PR against `context_prompts` in order to see the actual diff

stephmilovic committed:
wip

stephmilovic committed:
Merge branch 'context_prompts' into tile_prompts

stephmilovic committed:
more wip

stephmilovic committed:
ui works!

stephmilovic committed:
ui/api done

stephmilovic committed:
add feature flag

stephmilovic committed:
fix isKnowledgeBaseInstalled

stephmilovic committed:
Merge branch 'context_prompts' into tile_prompts

stephmilovic committed:
fix type

stephmilovic committed:
rm outdated comment

stephmilovic committed:
move check within docs.length 0

stephmilovic committed:
fix type in DQD code

stephmilovic committed:
fix types

stephmilovic committed:
add connector_id to request

stephmilovic committed:
add security_labs_tool tests

stephmilovic committed:
fix tests

KDKHD commented on x-pack/platform/packages/shared/kbn-elastic-assistant/impl/assistant/assistant_body/starter_prompts.tsx:
```diff
@@ -0,0 +1,147 @@
+};
+
+export const getAllPromptIds = (pGroups: PromptGroup[]) => {
+  return pGroups.map((promptGroup: PromptGroup) => [...Object.values(promptGroup)]).flat();
```
flatMap could be used

KDKHD commented on x-pack/solutions/security/plugins/security_solution/public/assistant/use_assistant_availability/index.tsx:
```diff
@@ -46,12 +31,17 @@ export const useAssistantAvailability = (): UseAssistantAvailability => {
     capabilities.actions?.delete === true &&
     capabilities.actions?.save === true;
 
+  const starterPromptsEnabled = useIsExperimentalFeatureEnabled('starterPromptsEnabled');
```
Is this feature flag still enabled through xpack.securitySolution.enableExperimental or is this the new feature flag service?

KDKHD commented on x-pack/solutions/security/plugins/security_solution/public/assistant/use_assistant_availability/index.tsx:
```diff
@@ -46,12 +31,17 @@ export const useAssistantAvailability = (): UseAssistantAvailability => {
 
+  const starterPromptsEnabled = useIsExperimentalFeatureEnabled('starterPromptsEnabled');
+  // remove once product has signed off on prompt text
+  const isStarterPromptsEnabled = starterPromptsEnabled;
```
Extra re-assignment?

KDKHD commented on x-pack/solutions/security/plugins/security_solution/server/assistant/tools/security_labs/security_labs_tool.test.ts:
```diff
@@ -100,5 +101,23 @@ In previous publications,`,
+
+      const result = await tool.func({ query: 'What is Kibana Security?', product: 'kibana' });
+
+      expect(result).toContain('The "AI Assistant knowledge base" needs to be installed');
```
Nice!

KDKHD commented on x-pack/solutions/security/plugins/security_solution/server/assistant/tools/security_labs/security_labs_tool.ts:
```diff
@@ -52,6 +53,14 @@ export const SECURITY_LABS_KNOWLEDGE_BASE_TOOL: AssistantTool = {
+          const isKnowledgeBaseInstalled = await getIsKnowledgeBaseInstalled(kbDataClient);
+          if (!isKnowledgeBaseInstalled) {
+            // prompt to help user install knowledge base
+            return 'The "AI Assistant knowledge base" needs to be installed, containing the Security Labs content. Navigate to the Knowledge Base page in the AI Assistant Settings to install it.';
```
```suggestion
            return 'The "AI Assistant knowledge base" needs to be installed, this will install the Security Labs content. Navigate to the Knowledge Base page in the AI Assistant Settings to install it.';
```

KDKHD reviewed (commented):
Code changes look good - How do I test the starter prompts?

stephmilovic committed:
Merge branch 'context_prompts' into tile_prompts

JiaweiWu committed:
[Response Ops][Task Manager] Skip API Key Tasks If Security is Disabled  (#221732)

## Summary

Resolves: https://github.com/elastic/kibana/issues/216810

This PR uses the licensing plugin to check is security is enabled. If
not, we skip adding API keys to tasks that pass in a `request` object,
but still scheduling them.

### Checklist
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

stephmilovic committed:
Merge branch 'context_prompts' into tile_prompts

stephmilovic committed:
[Security Assistant] Context prompts (#224956)

stephmilovic committed:
Merge branch 'main' into tile_prompts

stephmilovic committed:
change feature flag

stephmilovic committed:
rm ff value from security_solution
